### PR TITLE
Fix #28147

### LINKED ISSUE #28147: "default member initializer required" error in building C++ v1.42.0
### What version of gRPC and what language are you using?
C++ v1.42.0

### What operating system (Linux, Windows,...) and version?
Linux debian buster

### What runtime / compiler are you using (e.g. python version or version of gcc)
bazel version 4.0.0
gcc/g++ version 8.3.0

### What did you do?
Update gRPC version from v1.41.0 to v1.42.0 for existing code.

### What did you expect to see?
No error.

### What did you see instead?
Compile error on this line: https://github.com/grpc/grpc/blob/25446c468a81d74316fc833877cd0754ab2a9a72/src/core/ext/xds/xds_api.h#L58

Ref: https://stackoverflow.com/q/53408962

Error message:
/usr/include/c++/9/variant:275:5:   required from 'constexpr const bool std::__detail::__variant::_Traits<grpc_core::XdsApi::Route::UnknownAction, grpc_core::XdsApi::Route::RouteAction, grpc_core::XdsApi::Route::NonForwardingAction>::_S_copy_ctor'
/usr/include/c++/9/variant:1220:11:   required from 'class std::variant<grpc_core::XdsApi::Route::UnknownAction, grpc_core::XdsApi::Route::RouteAction, grpc_core::XdsApi::Route::NonForwardingAction>'
external/com_github_grpc_grpc/src/core/ext/xds/xds_api.h:181:68:   required from here
/usr/include/c++/9/type_traits:883:12: error: default member initializer for 'grpc_core::XdsApi::Duration::seconds' required before the end of its enclosing class
  883 |     struct is_constructible
      |            ^~~~~~~~~~~~~~~~
In file included from external/com_github_grpc_grpc/src/core/ext/xds/xds_certificate_provider.h:24,
                 from external/com_github_grpc_grpc/src/core/ext/xds/xds_certificate_provider.cc:21:
external/com_github_grpc_grpc/src/core/ext/xds/xds_api.h:56:21: note: defined here
   56 |     int64_t seconds = 0;
      |                     ^~~~
In file included from /usr/include/c++/9/bits/move.h:55,
                 from /usr/include/c++/9/bits/stl_pair.h:59,
                 from /usr/include/c++/9/bits/stl_algobase.h:64,
                 from /usr/include/c++/9/bits/stl_tree.h:63,
                 from /usr/include/c++/9/set:60,
                 from external/com_github_grpc_grpc/src/core/ext/xds/xds_api.h:26,
                 from external/com_github_grpc_grpc/src/core/ext/xds/xds_certificate_provider.h:24,
                 from external/com_github_grpc_grpc/src/core/ext/xds/xds_certificate_provider.cc:21:
/usr/include/c++/9/type_traits:883:12: error: default member initializer for 'grpc_core::XdsApi::Duration::nanos' required before the end of its enclosing class
  883 |     struct is_constructible
      |            ^~~~~~~~~~~~~~~~
In file included from external/com_github_grpc_grpc/src/core/ext/xds/xds_certificate_provider.h:24,
                 from external/com_github_grpc_grpc/src/core/ext/xds/xds_certificate_provider.cc:21:
external/com_github_grpc_grpc/src/core/ext/xds/xds_api.h:57:19: note: defined here
   57 |     int32_t nanos = 0;
      |                   ^~~~
INFO: Elapsed time: 267.893s, Critical Path: 6.09s
INFO: 825 processes: 7 internal, 818 linux-sandbox.
FAILED: Build did NOT complete successfully
FAILED: Build did NOT complete successfully
Error: Process completed with exit code 1.

### Anything else we should know about your project / environment?
N/A

### PR DESCRIPTION
**Pull request by owent**
According to my test and https://stackoverflow.com/questions/53408962/try-to-understand-compiler-error-message-default-member-initializer-required-be , `Duration() = default;` can not solve #28147

@drfloob

### CODE REVIEW

File: src/core/ext/xds/xds_common_types.h [hunk 1]
```diff
@@ -36,7 +36,7 @@ struct Duration {
   int64_t seconds = 0;
   int32_t nanos = 0;
 
-  Duration() = default;
+  Duration() : seconds(0), nanos(0) {}
```
Comment (mwtian) [inline src/core/ext/xds/xds_common_types.h:39]: Drive-by comment: it seems the code here is recent, not in 1.42. Maybe just `Duration() {}` since seconds and nanos are already default initialized?

Comment (owent) [reply src/core/ext/xds/xds_common_types.h:39]: There is no constructor for Duration on v1.42.0. But I have test using `Duration() = default` still has the problem of #28147 .
I have tested `Duration() : seconds(0), nanos(0) {}` in my case and it works. But I'm not sure if it will produce the correct codes when using `Duration(){}` with initializing members in declaration.
I need more suggestion here and thanks.

**Comment by owent** [comment]
This is already solved by another PR
